workers/worker_scheduler.rb

```diff
@@ -25,11 +32,25 @@ def initialize(sleep_interval: nil, cron_schedule: nil)
     end
 
     def schedule(&block)
-      if !self.sleep_interval.nil?
+      if self.sleep_interval
         block.call
         Kernel.sleep(self.sleep_interval)
-      elsif !self.cron_schedule.nil?
-        self.scheduler.cron(self.cron_schedule) { block.call }
+      elsif self.cron_schedule && self.scheduled_cron_job.nil?
+        job_id = self.scheduler.cron(self.cron_schedule) do
+          self.scheduled_cron_job = nil
```

Comment:
Here is the cron loop, right? As in, every time the cron job is run, this block is executed?